File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_max = -0x69696cafebabe
        new_max = -0x69696cafebabe
        
        for i in nums:
            new_max = max(i, new_max + i)
            if new_max > current_max:
                current_max = new_max
        return current_max

File: maximum-subarray/maximum-subarray.java
import java.lang.Math;

class Solution {
    public int maxSubArray(int[] nums) {
        int out = Integer.MIN_VALUE;
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            out = Math.max(out, sum);
            if (sum < 0) sum = 0;
        }
        return out;
    }
}